python/graphstorm/inference/emb_infer.py

```diff
@@ -0,0 +1,106 @@
+                                BUILTIN_TASK_LINK_PREDICTION)
+from .graphstorm_infer import GSInferrer
+from ..model.utils import save_embeddings as save_gsgnn_embeddings
+from ..model.gnn import do_full_graph_inference, do_mini_batch_inference
+from ..utils import sys_tracker, get_rank, get_world_size, barrier
+
+
+class GSgnnEmbGenInferer(GSInferrer):
+    """ Embedding Generation inferrer.
+
+    This is a high-level inferrer wrapper that can be used directly
+    to generate embedding for inferer.
+
+    Parameters
+    ----------
+    model : GSgnnNodeModel
+        The GNN model with different task.
+    """
+    def infer(self, data, task_type, save_embed_path, eval_fanout,
+            use_mini_batch_infer=False,
+            node_id_mapping_file=None,
+            save_embed_format="pytorch"):
+        """ Do Embedding Generating
+
+        Generate node embeddings and save into disk.
+
+        Parameters
+        ----------
+        data: GSgnnData
+            The GraphStorm dataset
+        task_type : str
+            task_type must be one of graphstorm builtin task types
+        save_embed_path : str
+            The path where the GNN embeddings will be saved.
+        use_mini_batch_infer : bool
+            Whether to use mini-batch inference when computing node embeddings.
+        node_id_mapping_file: str
+            Path to the file storing node id mapping generated by the
+            graph partition algorithm.
+        save_embed_format : str
+            Specify the format of saved embeddings.
+        """
+
+        device = self.device
+
+        assert save_embed_path is not None, \
+            "It requires save embed path for gs_gen_node_embedding"
+
+        sys_tracker.check('start embedding generation')
+        self._model.eval()
+
+        if task_type == BUILTIN_TASK_LINK_PREDICTION:
+            infer_ntypes = None
+        elif task_type in {BUILTIN_TASK_NODE_REGRESSION, BUILTIN_TASK_NODE_CLASSIFICATION}:
+            infer_ntypes = sorted(data.infer_idxs)
+        elif task_type in {BUILTIN_TASK_EDGE_CLASSIFICATION, BUILTIN_TASK_EDGE_REGRESSION}:
+            infer_ntypes = set()
+            for etype in data.infer_idxs:
+                infer_ntypes.add(etype[0])
+                infer_ntypes.add(etype[2])
+            infer_ntypes = sorted(infer_ntypes)
+        else:
+            raise TypeError("Not supported for task type: ", task_type)
+
+        if use_mini_batch_infer:
+            embs = do_mini_batch_inference(self._model, data, fanout=eval_fanout,
+                                           edge_mask=None,
+                                           task_tracker=self.task_tracker,
+                                           infer_ntypes=infer_ntypes)
+        else:
+            embs = do_full_graph_inference(self._model, data, fanout=eval_fanout,
+                                           edge_mask=None,
+                                           task_tracker=self.task_tracker)
```

Comment:
if infer_ntypes is not None, we only save embs in infer_ntypes, even it is full graph inference.